Library/brew.sh

```diff
@@ -230,6 +230,15 @@ then
     & disown
 fi
 
+if [[ -n "$HOMEBREW_AUTO_UPDATE" ]] &&
+   [[ "$HOMEBREW_COMMAND" = "install" || "$HOMEBREW_COMMAND" = "upgrade" ]]
+then
+  # Hide shellcheck complaint:
+  # shellcheck source=/dev/null
+  source "$HOMEBREW_LIBRARY/Homebrew/cmd/update.sh"
+  HOMEBREW_PRECOMMAND="homebrew-update --preinstall"
+fi
```

Comment:
> Does that still avoid the "file updated under it" issue, yeh?

Yes, it does. The function has already been completely parsed by the time it is executed.